detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenPublicDataClassSpec.kt

```diff
@@ -0,0 +1,147 @@
+        }
+
+        it("private data class should pass") {
+            val code = """
+                private data class C(val a: String)                
+            """
+
+            assertThat(subject.compileAndLint(code)).isEmpty()
+        }
+
+        it("internal data class should pass") {
+            val code = """
+                internal data class C(val a: String)                
+            """
+
+            assertThat(subject.compileAndLint(code)).isEmpty()
+        }
+
+        it("public class should pass") {
+            val code = """
+                class C(val a: String) 
+            """
+
+            assertThat(subject.compileAndLint(code)).isEmpty()
+        }
+
+        it("private data class inside a public class should pass") {
+            val code = """
+                class C {
+                    private data class D(val a: String)   
+                }
+            """
+
+            assertThat(subject.compileAndLint(code)).isEmpty()
+        }
+
+        it("public data class inside a public class should fail") {
+            val code = """
+                class C {
+                    data class D(val a: String)   
+                }
+            """
+
+            assertThat(subject.compileAndLint(code)).hasSize(1)
+        }
+
+        it("protected data class inside a public class should fail") {
+            val code = """
+                open class C {
+                    protected data class D(val a: String)   
+                }
+            """
+
+            assertThat(subject.compileAndLint(code)).hasSize(1)
+        }
+
+        it("public data class inside an internal class should pass") {
+            val code = """
+                internal class C {
+                    data class D(val a: String)   
+                }
+            """
+
+            assertThat(subject.compileAndLint(code)).isEmpty()
+        }
+
+        it("public data class inside an internal package should pass") {
+            val code = """
+                package com.example.internal
+                
+                data class C(val a: String)                
+            """
+
+            assertThat(subject.compileAndLint(code)).isEmpty()
+        }
+
+        it("public data class inside an internal subpackage should pass") {
+            val code = """
+                package com.example.internal.other
+                
+                data class C(val a: String)                
+            """
+
+            assertThat(subject.compileAndLint(code)).isEmpty()
+        }
+
+        it("public data class inside an internalise package should fail") {
+            val code = """
+                package com.example.internalise
+                
+                data class C(val a: String)                
+            """
+
+            assertThat(subject.compileAndLint(code)).hasSize(1)
+        }
+
+        it("public data class inside a random should fail") {
```

Comment:
internal package

Reply:
If you point to this line I have no idea what are you mean. If you talk about "internalise" it's not a typo. I want to check that a package that starts with internal but it's not just internal is not treated as an exception.

Reply:
Sorry for the confusion

```suggestion
        it("public data class inside a random package should fail") {
```